components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/LegacyJobOrchestrator.cs

```diff
@@ -171,7 +171,12 @@ public class LegacyJobOrchestrator : IJobOrchestrator {
                 }
             }
             else {
-                RefreshJobFromFile();
+                //avoid double events from FileSystemWatcher
+                DateTime lastWriteTime = File.GetLastWriteTime(_legacyCliModel.PublishedNodesFile);
+                if (lastWriteTime.Ticks - _lastRead.Ticks > _interval10ms) {
```

Comment:
Can we please use difference of `DateTime` objects and compare that with a `TimeSpan` object, something like `TimeSpan.FromMilliseconds(10)` ? Using `Ticks` is quite confusing as they represent hundred nanoseconds, so having interval of `10ms` be equal to `100000` looks strange because of that.

Reply:
Also, what if the file has actually changed within last `10ms` ? Then we would ignore those changes. I would recommend to ignore refresh only if we are sure that file hash is the same.

Reply:
changed using `TimeSpan.FromMilliseconds(10)` instead of ticks.
If I just ignore refresh when the file hash are the same I miss the event when you re-save the file for example (file overwritten with same hash). 
You are right that some events could be missed in theory, but I don't think we have two file changes in 10 ms

Reply:
@dacolgit I agree with @karok2m we should move the code into the method `RefreshJobFromFile` in the case were we know that the hashes are identically. And I would increase the value for 10 seconds.

Reply:
I moved the logic to `RefreshJobFromFile`.
Regarding increasing the debounce time, the duplicated event is fired just few ticks after the first one. This is a known issue of  FileSystemWatcher. Increasing the debounce to 10 seconds can lead to missing some entries in the log (for example if you re-save publishednodes.json with no changes within a 10 seconds interval)

Reply:
OK